ui/src/app/job-list/job-list.component.ts

```diff
@@ -75,6 +81,18 @@ export class JobListComponent implements OnInit {
         this.handleError(event.error);
       }
     });
+
+    // set project ID (if any) and get display field info for list columns
+    const req = URLSearchParamsUtils.unpackURLSearchParams(this.route.snapshot.queryParams['q']);
+    this.projectId = req.extensions.projectId || '';
+    const savedColumnSettings = this.settingsService.getDisplayColumns(this.projectId);
+    let field: DisplayField;
+    this.capabilities.displayFields.forEach((df) => {
+      field = df;
+      field.primary = true;
```

Comment:
mutating the original capabilities DisplayField object does seem like it may eventually lead to bugs, e.g. if we ever wanted to have a "reset to defaults" option on the column settings. I would either fix or at least note this in a comment for now though.